### ISSUE TITLE
Questions about documentation

### ISSUE THREAD

**Issue by datarttu** [datarttu/sujuikoDB#17]
Let's assume there's a new developer / GIS analyst who should be able to learn about this project and eventually contribute to it. OR there is a developer in another project helping to integrate our and their work. What is important from their perspective?

- What would be the best way to organize the documentation? Where? (Scope: inside this repo)
  - What should be included in the root README?
  - Shall we attach more specific readme files to each context like now (e.g. network readme lives in the same directory as the `nw` schema scripts), or collect them into a separate `docs` or `wiki` directory?
  - Where should pics and other files linked to readme files be located?
- Shall we make use of Postgres `COMMENT ON` feature (all the way from procedures up to individual columns) or document key things about relations, views, functions and procedures elsewhere?
- How much and how detailed usage examples should we provide vs. relying on a newcomer to clone the repo and try out things themself? Do we have to demonstrate e.g. what every essential relation looks like with a couple of rows? (-> the doc would become quite a wall of text)

**Comment by haphut** [comment]
You've done great work! That's a commendable amount of documentation. And I find the root `README.md` beautiful.

I would prioritize documentation issues like so:
1. Enough of it
1. Up-to-date
1. Easy to find what one is searching for

## Enough of it

Unless you know that some important parts are completely in the dark, there is enough documentation at this point that it seems like a task for the other team members to assess the coverage by trying it out.

A few thoughts:
- I would add more detail to the deployment instructions which now has the **TODO** note in it.
- Maybe rename the section to "Development".
- This is a crucial section for team members.
- If there were small test data sets also for HFP in the repo or behind links, that would be great. Maybe I missed them in my first cursory check.

## Up-to-date

I suggest that you add keeping the documentation up-to-date into the Definition of Done (DoD). E.g. when you change the code, the documentation must be updated correspondingly before the task is considered finished.

## Easy to find what one is searching for

The challenge becomes finding the right document if they are spread all over.

Crosslinking the documents heavily is important. Not only does it help finding relevant information but it gives the reader a sense that they know the topic once they see links to the same documents again and again.

I prefer to wait quite long before splitting a long document into smaller ones. Text search within the same document in the browser is effortless. GitHub also allows linking to Markdown sections.

I really like your approach of `README.md` files in separate directories. It gives you space to explain pieces of the architecture or the data model in its context.

As the project also has the directory `docs/`, it becomes important to clarify what to find in `docs/` and what to find elsewhere.

Unless there's a need for marketing the project, I would wait quite a bit before creating a website just for documentation.

A few options off the top of my head:
- Only information general, common or shared enough lives in `docs/`. Link heavily.
- Almost all documentation lives in `docs/`. Keep other `README.md`s sparse.

A few remaining thoughts:
- Maybe move the links to the data sources right next to the deployment section and refer to it from the deployment advice.
- Adding a Table of Contents (ToC) helps to get the big picture in large documents. You might reach for `npx doctoc` if necessary. I would not bother with ToCs for short documents unless you wish for a uniform presentation.

## Your questions

- Scope-wise the root `README.md` looks fine at first glance.
- The images are fine where they are. Keep using the directory consistently.
- I would use `COMMENT ON` if some API generation uses them. Otherwise I might rather explain things in Markdown and with images.
- I think it's a bit too much to document `SELECT * ... LIMIT 3` for every table. If there's demand for it, one could generate those automatically for every commit.

## Bottom line

Before I get deeper into documentation, it looks to be in a fine shape.

**Comment by datarttu** [comment]
Thank you for an extensive answer! I consider this issue closed and list the following things for myself to keep in mind with this project:

- Update README Deployment section (#29 and #30) and also think about the data source descriptions (#31)
- Separate context-specific `README.md` files are OK, keep on using them for now and link to them from the root README when needed. `docs/` is primarily for images and their sources (e.g. drawio files).
- Consider describing functions, relations and fields (at least non-trivial ones) in README instead of `COMMENT ON`, as we are probably not going to use an API that would make use of that feature, for example.
- Think about keeping documentation up-to-date in terms of DoD (#28)
- Include test data in the repository (#22 #23 #24 #25 #26 #27)
